### PR TITLE
Convert regular comments to doxygen-style comments

### PR DESCRIPTION
**Pull request by deepskyblue86**
Signed-off-by: Angelo Puglisi <angelopuglisi86@gmail.com>

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

/kind documentation

> /kind failing-test

> /kind feature

> If contributing rules or changes to rules, please make sure to also uncomment one of the following line:

> /kind rule-update

> /kind rule-create

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

> /area engine

> /area rules

> /area tests

> /area proposals

**What this PR does / why we need it**:
Converting simple comments do doxygen-style comments.
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
docs: development docs for the Falco engine 
```

### CODE REVIEW

**Comment by deepskyblue86** [comment]
Preview of `falco_engine`:
image

File: userspace/engine/falco_engine.h [hunk 1]
```diff
@@ -38,118 +38,128 @@ limitations under the License.
...
 #include "falco_common.h"
 
-//
-// This class acts as the primary interface between a program and the
-// falco rules engine. Falco outputs (writing to files/syslog/etc) are
-// handled in a separate class falco_outputs.
-//
-
+/**
+ * @brief The primary interface between a program and the falco rules engine.
```
Comment (leodido) [inline userspace/engine/falco_engine.h:42]:
```suggestion
 * @brief The primary interface between a program and the Falco rules engine.
```

File: userspace/engine/falco_engine.h [hunk 2]
```diff
@@ -38,118 +38,128 @@ limitations under the License.
...
+	 * @note If you provide any ruleset to @ref enable_rule/@ref enable_rule_by_tag,
+	 * you should look up the ruleset id and pass it to
+	 * @ref process_sinsp_event/@ref process_k8s_audit_event.
+	 */
 	uint16_t find_ruleset_id(const std::string &ruleset);
 
-	//
-	// Return the number of falco rules enabled for the provided ruleset
-	//
+	//! Return the number of falco rules enabled for the provided ruleset
```
Comment (leodido) [inline userspace/engine/falco_engine.h:128]:
```suggestion
	//! Return the number of Falco rules enabled for the provided ruleset
```

**Review by leodido — APPROVED** [review]
Grazie Angelo!

✅ 

putting this on
/hold
until the PR on which branch it bases on is merged in

**Review by leogr — APPROVED** [review]
/approve

**Comment by deepskyblue86** [comment]
Note that this PR depends on #1513

**Pull request closed without merging**